### PR TITLE
Change doors to default.can_interact_with_node()

### PR DESCRIPTION
**Pull request by Thomas--S**
Why?
* Reduces code duplications
* Makes door protection override-able (see #1559)

### CODE REVIEW

**Comment by sofar** [comment]
instead of an LBM, can you convert the meta on access?

**Comment by Thomas--S** [comment]
I could do this, however, I actually think it would be better if this conversion happens on a central place, so that you can rely on finding the owner in "owner" instead of "doors_owner". However, if there is heavy disagreement for this approach, I can also change it.

**Comment by sofar** [comment]
LBM conversions may work well, but they could be messed up by schematic placement and code that manually sets `doors_owner`. It's always better to convert the data on use than on block load (once).

**Comment by Thomas--S** [comment]
@sofar PR updated according to your suggestions.

**Comment by sofar** [comment]
Still not happy with the LBM, I feel like it's almost less code to just test-and-convert when needed and adding LBM's for metadata stuff seems a bit over the top.

**Comment by Thomas--S** [comment]
I added the LBM for other mods that want to access the owner field without backwards-compatibility code. If you want the LBM to be removed, I can do this, too.

**Comment by Thomas--S** [comment]
@sofar I removed the LBM now. I ask you to resolve the merge conflicts. You have write access to this PR. Thanks!

**Comment by paramat** [comment]
I assume you're not able to rebase yourself? If devs are happy with the code you could always open a fresh PR to solve the conflicts.

**Comment by Thomas--S** [comment]
I think the rebase itself wouldn't be the problem.
The problem is, that I didn't have the time yet to understand the code of the conflicting commit. Maybe I have time to look at this tomorrow.

**Comment by Thomas--S** [comment]
Rebased and updated to new requirements.
I tested this and it should work. I did not test edge-cases.

File: mods/doors/init.lua [hunk 1]
```diff
@@ -394,7 +385,7 @@ function doors.register(name, def)
 		end
 		def.on_skeleton_key_use = function(pos, player, newsecret)
 			local meta = minetest.get_meta(pos)
-			local owner = meta:get_string("doors_owner")
+			local owner = meta:get_string("owner")
```
Comment (SmallJoker) [inline mods/doors/init.lua]: What about backwards compatibility here?

Comment (Thomas--S) [inline mods/doors/init.lua]: Thanks, I have forgotten this one.

**Comment by Thomas--S** [comment]
@paramat @rubenwardy @sofar Could you please review this?

**Comment by Thomas--S** [comment]
Rebased again.
I tested this for a short time again, it seems to still work fine.
I hope this gets reviewed soon.
I really don't want to rebase it a third time.

File: mods/doors/init.lua [hunk 2]
```diff
@@ -6,6 +6,17 @@ local _doors = {}
 _doors.registered_doors = {}
 _doors.registered_trapdoors = {}
 
+local function replace_old_owner_information(pos)
+	local meta = minetest.get_meta(pos)
+	local owner = meta:get_string("doors_owner")
+	if not owner or owner == "" then
```
Comment (octacian) [inline mods/doors/init.lua]: It would be easier to understand here if you formatted the code as below:

```lua
if owner and owner ~= "" then
    meta:set_string("owner", owner)
    meta:set_string("doors_owner", "")
end
```

Not only does it shorten the code by removing the need for the `return` and `else`, it also makes it more readably. Aside from that, :+1:

Comment (Thomas--S) [inline mods/doors/init.lua]: Thanks!

**Comment by Thomas--S** [comment]
Thank you very much for merging!

**Comment by mase76** [comment]
I am not able to open my wooden doors any more.

**Comment by Thomas--S** [comment]
#1649
